experimental/dds/tree/src/generic/GenericSharedTree.ts

```diff
@@ -303,24 +305,13 @@ export abstract class GenericSharedTree<TChange> extends SharedObject<ISharedTre
 	}
 
 	/**
-	 * {@inheritDoc @fluidframework/shared-object-base#SharedObject.snapshotCore}
+	 * {@inheritDoc @fluidframework/shared-object-base#SharedObject.summarizeCore}
 	 */
-	public snapshotCore(serializer: IFluidSerializer): ITree {
-		const tree: ITree = {
-			entries: [
-				{
-					mode: FileMode.File,
-					path: snapshotFileName,
-					type: TreeEntry[TreeEntry.Blob],
-					value: {
-						contents: this.saveSerializedSummary({ serializer }),
-						encoding: 'utf-8',
-					},
-				},
-			],
-		};
-
-		return tree;
+	public summarizeCore(serializer: IFluidSerializer, fullTree: boolean): ISummaryTreeWithStats {
+		const blobContent = this.saveSerializedSummary({ serializer });
+		const builder = new SummaryTreeBuilder();
+		builder.addBlob(snapshotFileName, blobContent);
+		return builder.getSummaryTree();
```

Comment:
Given that same sequence of creation single-blob summary repeats in this PR, maybe create a helper function for that?
I.e., this code to look something like
```
return createSingleBlobSummary(snapshotFileName, this.saveSerializedSummary({ serializer }));
```